aks-node-controller/app.go

```diff
@@ -65,6 +67,9 @@ func (a *App) run(ctx context.Context, args []string) error {
 	case "provision-wait":
 		provisionStatusFiles := ProvisionStatusFiles{ProvisionJSONFile: provisionJSONFilePath, ProvisionCompleteFile: provisionCompleteFilePath}
 		provisionOutput, err := a.ProvisionWait(ctx, provisionStatusFiles)
+		if err != nil {
+			slog.Error("provision-wait failed", "error", err)
+		}
```

Comment:
Usually I avoid logging error and returning it at the same time. Typically it just lead to logging same error twice.